codec-framework/src/test/java/org/interledger/encoding/asn/codecs/AsnUintCodecTest.java

```diff
@@ -0,0 +1,111 @@
+package org.interledger.encoding.asn.codecs;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import com.google.common.io.BaseEncoding;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+
+@RunWith(Parameterized.class)
+public class AsnUintCodecTest {
+
+  private final BigInteger expectedUint;
+  private final byte[] exptedEncodedBytes;
+
+  private AsnUintCodec codec;
+
+  /**
+   * Construct an instance of this parameterized test with the supplied inputs.
+   *
+   * @param expectedUint       The expected value, as a {@link BigInteger}, of {@code
+   *                           exptedEncodedBytes}, once encoded.
+   * @param exptedEncodedBytes The expected encoded value, in bytes, of {@code expectedUint}.
+   */
+  public AsnUintCodecTest(
+      final BigInteger expectedUint, final byte[] exptedEncodedBytes) {
+    this.expectedUint = Objects.requireNonNull(expectedUint);
+    this.exptedEncodedBytes = Objects.requireNonNull(exptedEncodedBytes);
+  }
+
+  /**
+   * The data for this test...
+   */
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {
+            BigInteger.ZERO,
+            BaseEncoding.base16().decode("00")
+        },
+        {
+            BigInteger.valueOf(1L),
+            BaseEncoding.base16().decode("01")
+        },
+        {
+            BigInteger.TEN,
+            BaseEncoding.base16().decode("0A")
+        },
+        {
+            BigInteger.valueOf(15L),
+            BaseEncoding.base16().decode("0F")
+        },
+        {
+            BigInteger.valueOf(255L),
+            BaseEncoding.base16().decode("FF")
+        },
+        {
+            BigInteger.valueOf(1024L),
+            BaseEncoding.base16().decode("0400")
+        },
+        {
+            BigInteger.valueOf(16385L),
```

Comment:
Why not use hex literals ? 0x4001L etc ?

Reply:
Good question - I think the test is clearer if we use Base10 numbers here because the purpose of this test is to ensure that a `BigInteger` will encode into binary. Since `BigInteger` is always operated upon in Base10, I want to be sure that the B10 number properly encodes/decodes. So, the `expectedUint` value is always represented in Base10 on purpose, and we only use the HEX value to construct the byte-array.

Reply:
Actually, looking more closely at this test, @adrianhopebailie, this seems to be treating this like a Uint64, but instead the `AsnUintCodec` should be using a length-prefix, no? (since the number of octets can be arbitrary)

Reply:
I think the serializer classes handle that right ? (checking ... )

Reply:
Hrmmm, this codec doesn't actually seem to be registered at all for the OER rules?

Reply:
re: hex literals, I just thought they'd more easily eyeball match the hex encoding

Reply:
@sappenin 

Yeah, it seems I recalled correctly, and that the serializers handle all the length prefix encoding. 

See:
https://github.com/hyperledger/quilt/blob/b53a0511007c005c1f5455ec7f59639fb935e5b6/codec-framework/src/main/java/org/interledger/encoding/asn/framework/CodecContextFactory.java#L65
https://github.com/hyperledger/quilt/blob/b53a0511007c005c1f5455ec7f59639fb935e5b6/codec-framework/src/main/java/org/interledger/encoding/asn/serializers/oer/AsnOctetStringOerSerializer.java#L88

Reply:
Oh dang, you're totally right. I've been spending so much time in the Codecs I forgot about the Serializers.

Kk - I think this PR is good to go.